core/src/modules/DynamicsModule/BBMDynamics.cpp

```diff
@@ -38,9 +41,24 @@ void BBMDynamics::setData(const ModelState::DataMap& ms)
     vice = ms.at(vName);
 
     // Set the data in the kernel arrays.
+    // Required data
     for (const auto& fieldName : namedFields) {
         kernel.setData(fieldName, ms.at(fieldName));
     }
+    // Data that can have a default value
+    for (const auto entry : defaultFields) {
+        // Directly add data that is supplied
+        const std::string& fieldName = entry.first;
+        if (ms.count(fieldName) > 0)
+            kernel.setData(fieldName, ms.at(fieldName));
+        // Fill data that is not supplied, masking if the mask is available
+        ModelArray data(entry.second.first);
+        data.resize();
+        // Fill the default value
+        data = entry.second.second;
+        // Mask the default data
+        kernel.setData(fieldName, mask(data));
```

Comment:
Shouldn't this bit be in an ``else`` clause?

Reply:
Yes, it should